docs/source/content/unsupervised/clustering.rst

```diff
@@ -0,0 +1,222 @@
+consistent result each time you run the code. The rest of the code is to
+display the final plot shown in *Figure 6*.
+
+.. figure:: _img/KMeans.png
+
+   **Figure 6. A final clustered data set**
+
+The clusters are color coded, the 'x's represent cluster centers, and the
+dotted lines represent cluster boundaries.
+
+Hierarchical
+============
+Hierarchical clustering imagines the data set as a hierarchy of clusters. We
+could start by making one giant cluster out of all the data points. This is
+illustrated in *Figure 7*.
+
+.. figure:: _img/Hierarchical_Step1.png
+
+   **Figure 7. One giant cluster in the data set***
+
+Inside of this cluster, we find the two least similar sub-clusters and split
+them. This can be done by using an algorithm to maximize the inter-cluster
+distance. This is just the smallest distance between a node from one cluster
+and a node from the other cluster. This is illustrated in *Figure 8*.
+
+.. figure:: _img/Hierarchical_Step2.png
+
+   **Figure 8. The giant cluster is split into 2 clusters**
+
+We continue to split the sub-clusters until every data point belongs to its
+own cluster or until we decide to stop. If we start from one giant cluster and
+break it down into successively smaller clusters, it is called **top-down** or
+**divisive** clustering. Alternatively, we could start by considering a
+cluster for every data point. The next step would be to combine the two
+closest clusters into a larger cluster. This can be done by finding the
+distance between every cluster and choosing the pair with the least distance
+between them. We would continue this process until we had a single cluster.
+This method of combining clusters is called **bottom-up** or **agglomerative**
+clustering. At any point in these two methods, we can stop when the clusters
+look appropriate.
+
+Unlike K-Means, Hierarchical clustering is relatively slow so it doesn’t scale
+as well to large data sets. On the bright side, Hierarchical clustering is
+more consistent when you run it multiple times and doesn’t require you to know
+the number of expected clusters.
+
+The relevant code is available in the clustering_hierarchical.py_ file.
+
+.. _clustering_hierarchical.py: /code/unsupervised/Clustering/clustering_hierarchical.py
+
+In the code, we create the simple data set to use for analysis. Setting up the
+clustering is very simple and requires one line of code:
+
+.. code-block:: python
+
+   hierarchical = AgglomerativeClustering(n_clusters=3).fit(x)
+
+The `n_clusters` parameter was chosen to be 3 because there appears to be 3
+clusters in out data set. If we didn't already know this, we could try out
+different values and see which one worked the best. The rest of the code is to
+display the final plot shown in *Figure 9*.
+
+.. figure:: _img/Hierarchical.png
+
+   **Figure 9. A final clustered data set**
+
+The clusters are color coded and large clusters are surrounded with a border
+to show which data points belong to them.
+
+Summary
```

Comment:
The summary title should have '====' above also to keep it out of the Methods section